docs/src/content/docs/explainers/async-transform.mdx

```diff
@@ -1,6 +1,274 @@
+				// to JS
+				code: myExtToJs(src),
+				// if possible return a sourcemap to allow for better debugging
+				map: null,
+			}
+		}
+	}
+};
+`}/>
+
+In this case we were handling our own custom file extension and compile that to JS but you can also
+use the transform hook to change your JS.
+
+    <Code lang="ts" code={`
+
+export const simpleTransform = {
+	// name is used during debug
+	name: 'remove-console-logs',
+	transform(src){
+		return {
+			code: src.replaceAll(/console.log\(.+?\)/g, ""),
+			map: null,
+		}
+	}
+};
+`}/>
+
+This super simple plugin eliminate every instance of `console.log()` from your code.
+
+<Aside type="caution">
+	Don't use this anywhere it's just a silly example and you should probably be a bit more nuanced
+	than `replaceAll` if you want to remove logs from your code.
+</Aside>
+
+The gist of it is: you can do a lot of cool things with a Vite plugin and we decided to build the
+Async Transform to make your life (hopefully) easier!
+
+</details>
+
+## How the `Async Transform` changes my code?
+
+The reason why we are writing this explainer is because we firmly believe that if you allow you code
+to be changes by some script you should fully understand how it changes it and what the final result
+will look like.
+
+As shown in our [Mid run cancellation](/explainers/mid-run-cancellation) explainer this Vite plugin
+aim to solve a couple of problems with async code in Typescript:
+
+- Promises are non cancellable: once you start them there's no way to stop the code in the middle of
+  the function from executing
+- Generators can be "stopped" mid execution but Typescript doesn't play very well with the `yield`
+  keyword
+
+To fix this problems we let you write the simple and familiar async code and we provide a Vite
+plugin to transform your async functions into generators!
+
+<Tabs>
+
+<TabItem label="Before">
+
+```svelte
+<script lang="ts">
+	import { task } from '@sheepdog/svelte';
+
+	const myTask = task(async () => {
+		const result = await fetch('/api/my-endpoint');
+		return await result.json();
+	});
+</script>
+
+<button
+	on:click={() => {
+		myTask.perform();
+	}}
+>
+	perform
+</button>
+```
+
+</TabItem>
+
+<TabItem label="After">
+
+```svelte
+<script lang="ts">
+	import { task } from '@sheepdog/svelte';
+
+	const myTask = task(async function* () {
+		const result = yield fetch('/api/my-endpoint');
+		return yield result.json();
+	});
+</script>
+
+<button
+	on:click={() => {
+		myTask.perform();
+	}}
+>
+	perform
+</button>
+```
+
+</TabItem>
+
+</Tabs>
+
+As you can see we aim to touch your code the bare minimum. But I can still see the worry on your
+face so let's dive a bit more and let's make a couple of clarifications
+
+### What about my other async functions?
+
+You don't want to see all of your beloved async functions be turned into generators right? Well,
+worry not because we specifically target only the async functions that are arguments of our own
```

Comment:
That's personal feeling but this phrasing sounds slightly too familiar for me compared to the rest of the doc.

```suggestion
If you don't want to see all of your beloved async functions be turned into generators, worry not, because we specifically target only the async functions that are arguments of our own
```